car-service/src/main/java/com/ecommercepractice/carservice/service/ProductListService.java

```diff
@@ -0,0 +1,102 @@
+package com.ecommercepractice.carservice.service;
+
+import com.ecommercepractice.carservice.exception.ProductIdAlreadyInProductListException;
+import com.ecommercepractice.carservice.exception.ProductIdNotFoundedException;
+import com.ecommercepractice.carservice.exception.ProductListNotFoundedException;
+import com.ecommercepractice.carservice.model.ProductList;
+import com.ecommercepractice.carservice.model.ProductListId;
+import com.ecommercepractice.carservice.repository.ProductListRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class ProductListService {
+    @Autowired
+    ProductListRepository productListRepository;
+
+    /**
+     * Fetch all products associated to a product list Id
+     * @param idProductList
+     * @return
+     */
+    public List<ProductList> fetchById(Integer idProductList){
+        return productListRepository.fetchAllProductsWithIdProductList(idProductList)
+                .orElseThrow(() -> new  ProductListNotFoundedException(idProductList));
+    }
+
+    /**
+     * Add a product to a list
+     * @param idProductList
+     * @param productId
+     * @param quantity
+     */
+    public void appendProduct(Integer idProductList, Integer productId, Integer quantity){
```

Comment:
does this method should not increase the quantity of that product in the card instead of throws an exception?

Reply:
I consider it could lend to make mistakes as a User, I prefer to restring the method just to add products, changes about them must be via PUT.